### PR TITLE
Some more improvements

### PR DESCRIPTION
**Pull request by ghost**
Added even more keywords, added a lot of enumerated values, split block modifiers for a better use of 'disabling' chunks and enforce CamelCase syntax for better maintenance.

### CODE REVIEW

**Comment by kimburgess** [comment]
Nice! Thanks (again) for the contribution.

The addition syntax definitions look good, however I'm not sold on the enforced CamelCase. 

Personally I'm pretty OCD with adhering to that style, but the language is case insensitive. As a syntax parser this should match the language with stylistic enforcement left to a linter or other tools further up the chain.

**Comment by ghost** [comment]
Yeah, I guess you're right. I put the case-insensitive flag back in but left the wording. I don't have seen any sort of linter for ScalaScript

File: ScalaScript.sublime-syntax [hunk 1]
```diff
@@ -67,8 +78,24 @@ contexts:
       scope: constant.numeric.integer.scalascript
 
     # Booleans
-    - match: '(?i)\b(true|false|on|off)\b'
+    - match: '(?i)\b(True|False|On|Off)\b'
```
Comment (ghost) [inline ScalaScript.sublime-syntax:81]: I know the Designer converts boolean values true/false into On/Off on the next save...

Was there once the possibility to specifiy true/false explicitly in an older version. I only know about Scala Designer 6 and later.

File: ScalaScript.sublime-syntax [hunk 2]
```diff
@@ -51,9 +62,9 @@ contexts:
       scope: invalid.illegal.stray-bracket-end
 
     # Blocks
-    - match: '\{|(?i)\bevent\b'
+    - match: '\{|(?i)\bEvent\b'
       push: block
-    - match: '\}|(?i)\bend\b'
+    - match: '\}|(?i)\bEnd\b'
```
Comment (ghost) [inline ScalaScript.sublime-syntax:67]: I have never seen a block starting with Event and/or ending with End - is this still being used or is this deprecated?

Comment (kimburgess) [reply ScalaScript.sublime-syntax:67]: They're still [listed in the current docs](https://docs.scala.com/display/D1102/Lexical+Rules#LexicalRules-ControlPrimitives) as control primitives.

File: ScalaScript.sublime-syntax (hunk 1 above)
Comment (kimburgess) [reply ScalaScript.sublime-syntax:81]: Yep, still valid based on [the current syntax docs](https://docs.scala.com/display/D1102/Lexical+Rules#LexicalRules-BooleanLiterals)

**Pull request closed without merging**